src/execute-pr-actions.ts

```diff
@@ -22,21 +16,13 @@ const closePr = `mutation($input: ClosePullRequestInput!) { closePullRequest(inp
 const addProjectCard = `mutation($input: AddProjectCardInput!) { addProjectCard(input: $input) { clientMutationId } }`;
 const moveProjectCard = `mutation($input: MoveProjectCardInput!) { moveProjectCard(input: $input) { clientMutationId } }`;
 
-export async function executePrActions(actions: Actions) {
-    // Get the latest version of this PR's info
-    const info = await client.query<PRQueryResult>({
-        query: GetPRInfo,
-        variables: {
-            pr_number: actions.pr_number
-        },
-        fetchPolicy: "network-only",
-        fetchResults: true
-    });
-    const pr = info.data.repository?.pullRequest!;
-
-    const waiting: Promise<unknown>[] = [];
-
-    const labels = info.data.repository?.pullRequest?.labels?.nodes!;
+export async function executePrActions(actions: Actions, info: PRQueryResult, dry: true): Promise<string[]>;
+export async function executePrActions(actions: Actions, info: PRQueryResult, dry?: boolean): Promise<undefined>;
+export async function executePrActions(actions: Actions, info: PRQueryResult, dry?: boolean) {
```

Comment:
`info` is now reused from previous steps instead of re-fetched, and the `dry` switch builds mutation requests without executing them.